datafusion/sql/tests/sql_integration.rs

```diff
@@ -4487,6 +4490,87 @@ impl TableSource for EmptyTable {
     }
 }
 
+#[test]
+fn roundtrip_expr() {
+    let tests: Vec<(TableReference, &str, &str)> = vec![
```

Comment:
FYI @devinjdangelo  -- maybe we can consolidate the round trip expr tests from https://github.com/apache/arrow-datafusion/pull/9578 here too